source/ssb-chart/templates/access-ingress-internal-authorization-policy.yaml

```diff
@@ -0,0 +1,97 @@
+{{- if and .Values.access (and .Values.access.ingress .Values.access.ingress.internal) }}
+{{- range .Values.access.ingress.internal }}
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: "{{ template "app.name" $ }}-ingress-{{.namespace }}-{{ .application }}-auth-policy"
+  labels:
+{{ include "default.labels" $ | indent 4 }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "app.name" $ }}
+  action: ALLOW
+  rules:
+    {{- $fromPrincipal := printf "cluster.local/ns/%s/sa/%s-sa" .namespace .application }}
```

Comment:
The "-sa" suffix is a convention in this Helm chart. The suffix would most likely not be present for apps not deployed with this Helm chart. This makes it difficult opening ingress from other apps.

Not sure what is the best way to solve this, but something that might improve the situation is to remove the "-sa" suffix from the SA name in all templates and require that the SA name is used instead of the app name. For ssb-chart apps, the app and sa name would be the same  and we can still add rules for other apps. Update: One possible complication -> SA names referenced in Workload identities.

The suggested specification in the Jira task is a little lacking on this part.

Reply:
If the users should specify the service account we stand in the same "squeeze" when using network policy (where we use labels)

Reply:
I'll implement and draft for how this can be done, based on your suggestion

Reply:
I believe we add an "app:" label that is the same as the app name. The ideal would have been if SA name == app name == app label value. If we were using CalicoNetworkPolicy, we could have used SA in Network policies as well, but then it's probably better to start looking at Cilium or GKE Dataplane v2.
Let's just work with the "-sa" prefix for now to avoid scope creep.